packages/lumx-react/src/components/dialog/Dialog.tsx

```diff
@@ -58,8 +62,8 @@ export interface DialogProps extends GenericProps {
 
 export type DialogSizes = Extract<Size, 'tiny' | 'regular' | 'big' | 'huge'>;
 
-const isHeader = isComponent('header');
-const isFooter = isComponent('footer');
+const isHeader = (component: ReactNode) => isComponent('header')(component) || isComponent('DialogHeader')(component);
+const isFooter = (component: ReactNode) => isComponent('footer')(component) || isComponent('DialogFooter')(component);
```

Comment:
I'm not completely satisfied with this but I'm not sure how it could be written differently.
Do you have some lodash trickery or simply a better idea?